test-graphql/client-test.js

```diff
@@ -252,4 +254,38 @@ suite('client-test', () => {
       assert.ok(fetchMock.done());
     });
   });
+
+  test('it can fetch products with the query arg', () => {
+    const config = new Config({
+      domain: 'query-products.myshopify.com',
+      storefrontAccessToken: 'abc123'
+    });
+
+    const client = new Client(config);
+
+    fetchMock.postOnce('https://query-products.myshopify.com/api/graphql', {data: {shop: {products: {edges: [{node: {title: 'Cat'}}]}}}});
+
+    return client.fetchQueryProducts({title: 'Cat', limit: 10}, productConnectionQuery(['title'])).then((products) => {
```

Comment:
Do you think it'd also be useful to check the query sent to see if the query arg string is being built properly?